src/Core/Form/IdentifiableObject/DataHandler/CategoryFormDataHandler.php

```diff
@@ -200,4 +226,25 @@ private function uploadImages(
             }
         }
     }
+
+    /**
+     * @param int $categoryId
+     *
+     * @return array<int, int>
+     */
+    private function getAvailableKeys(int $categoryId): array
+    {
+        $files = scandir(_PS_CAT_IMG_DIR_, SCANDIR_SORT_NONE);
```

Comment:
- We should not use constants but DI instead.

- Maybe use the Symfony finder for this, it has a stable API and has better error management support.